### ISSUE TITLE
Threads don't seem to be cleaned up

### ISSUE THREAD

**Issue by daw3rd** [ibm-watson-iot/iot-java#94]
I have a web app that does
```
iotPlatform = new IOTPlatform(gatewayID, IOTPlatformProperties.instance().getProperties());	
try {
	iotPlatform.connect();
```
in the web app's contextInitialized() method and the following in contextDestroyed()
```
if (iotPlatform != null)
	iotPlatform.disconnect()
```
yet, I'm getting this when the server is shutdown.
```
Apr 11, 2017 2:17:03 PM org.apache.catalina.loader.WebappClassLoaderBase clearReferencesThreads
WARNING: The web application [IoT-AnomalyServer] appears to have started a thread named [MQTT Disc: g:sl0l32:CognitiveEarGW:CognitiveEar-9.76.28.149] but has failed to stop it. This is very likely to create a memory leak. Stack trace of thread:
 java.lang.Object.wait(Native Method)
 java.lang.Object.wait(Object.java:201)
 java.lang.Thread.join(Thread.java:722)
 java.lang.Thread.join(Thread.java:683)
 org.eclipse.paho.client.mqttv3.internal.CommsReceiver.stop(CommsReceiver.java:92)
 org.eclipse.paho.client.mqttv3.internal.ClientComms.shutdownConnection(ClientComms.java:335)
 org.eclipse.paho.client.mqttv3.internal.ClientComms$DisconnectBG.run(ClientComms.java:733)
 java.lang.Thread.run(Thread.java:785)
Apr 11, 2017 2:17:03 PM org.apache.catalina.loader.WebappClassLoaderBase clearReferencesThreads
WARNING: The web application [IoT-AnomalyServer] appears to have started a thread named [Timer-0] but has failed to stop it. This is very likely to create a memory leak. Stack trace of thread:
 java.lang.Object.wait(Native Method)
 java.lang.Object.wait(Object.java:201)
 java.util.TimerThread.mainLoop(Timer.java:563)
 java.util.TimerThread.run(Timer.java:516)
Apr 11, 2017 2:17:03 PM org.apache.catalina.loader.WebappClassLoaderBase clearReferencesThreads
WARNING: The web application [IoT-AnomalyServer] appears to have started a thread named [MQTT Snd: g:sl0l32:CognitiveEarGW:CognitiveEar-9.76.28.149] but has failed to stop it. This is very likely to create a memory leak. Stack trace of thread:
 java.lang.Object.wait(Native Method)
 java.lang.Object.wait(Object.java:172)
 org.eclipse.paho.client.mqttv3.internal.ClientState.get(ClientState.java:762)
 org.eclipse.paho.client.mqttv3.internal.CommsSender.run(CommsSender.java:101)
 java.lang.Thread.run(Thread.java:785)
Apr 11, 2017 2:17:03 PM org.apache.catalina.loader.WebappClassLoaderBase clearReferencesThreads
WARNING: The web application [IoT-AnomalyServer] appears to have started a thread named [MQTT Rec: g:sl0l32:CognitiveEarGW:CognitiveEar-9.76.28.149] but has failed to stop it. This is very likely to create a memory leak. Stack trace of thread:
 java.net.SocketInputStream.socketRead0(Native Method)
 java.net.SocketInputStream.socketRead(SocketInputStream.java:127)
 java.net.SocketInputStream.read(SocketInputStream.java:181)
 java.net.SocketInputStream.read(SocketInputStream.java:152)
 com.ibm.jsse2.a.a(a.java:134)
 com.ibm.jsse2.a.a(a.java:219)
 com.ibm.jsse2.as.a(as.java:648)
 com.ibm.jsse2.as.a(as.java:816)
 com.ibm.jsse2.e.read(e.java:30)
 com.ibm.jsse2.e.read(e.java:2)
 java.io.DataInputStream.readByte(DataInputStream.java:276)
 org.eclipse.paho.client.mqttv3.internal.wire.MqttInputStream.readMqttWireMessage(MqttInputStream.java:92)
 org.eclipse.paho.client.mqttv3.internal.CommsReceiver.run(CommsReceiver.java:116)
 java.lang.Thread.run(Thread.java:785)
```

**Comment by sathipal** [comment]
I would like to understand how this library is used internally? Also, doesn't see any of the library classes in the traces. Can you please elaborate?

**Comment by daw3rd** [comment]
I'm not doing much different than what I posted above.  I allocate and connect() the instance when the web app is loaded (in contextInitialized()) and disconnect() when the web app is shut down (in contextDestroyed()).  If the web app is used, then in between these calls I call iotPlatform.publish() from within the web app.  That's about it.  However, if I don't trigger the call to publish() then I don't get the errors above, but Tomcat still will not shutdown properly.  If I had to guess, I would say Paho or something is not stopping the threads it created.  One solution might be to make the threads daemons so they will die when the rest of the JVM is stopped.
